VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx

```diff
@@ -106,6 +113,55 @@ vtkMRMLVirtualRealityViewNode* qMRMLVirtualRealityHomeWidget::virtualRealityView
   return d->VirtualRealityViewNode;
 }
 
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityHomeWidget::setupModuleButton(QWidget* widget, QIcon* icon)
+{
+  Q_D(const qMRMLVirtualRealityHomeWidget);
+
+  QPushButton* moduleButton = new QPushButton(d->ModulesGroupBox);  
+  d->ModulesGroupBoxLayout->addWidget(moduleButton);
+  moduleButton->setIcon(*icon);
+  
+  widget->setParent(d->ModuleWidgetFrame);
+  this->ModuleWidgetsMap[moduleButton] = widget;
+  
+  QObject::connect(moduleButton, SIGNAL(clicked()), this, SLOT(onModuleButtonPressed()));
+  QObject::connect(d->backButton, SIGNAL(clicked()), this, SLOT(onBackButtonPressed()));
+ 
+  moduleButton->setVisible(true);
+  widget->setVisible(false);
+  d->backButton->setVisible(false);
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityHomeWidget::onModuleButtonPressed()
+{
+  Q_D(const qMRMLVirtualRealityHomeWidget);
+ 
+  QPushButton* moduleButton = qobject_cast<QPushButton*>(sender());
+  
+  d->HomeWidgetFrame->setVisible(false);
+  d->ModulesGroupBox->setVisible(false);
+  this->ModuleWidgetsMap[moduleButton]->setVisible(true);
+  d->backButton->setVisible(true);
+}
+
+//-----------------------------------------------------------------------------
+void  qMRMLVirtualRealityHomeWidget::onBackButtonPressed()
+{
+  Q_D(const qMRMLVirtualRealityHomeWidget);
+  d->HomeWidgetFrame->setVisible(true);
+  d->ModulesGroupBox->setVisible(true); 
+  d->backButton->setVisible(false);
+
+  QMap<QPushButton*, QWidget*>& map = this->ModuleWidgetsMap;
+  QMap<QPushButton*, QWidget*>::const_iterator i; 
+  for (i = map.constBegin(); i != map.constEnd(); ++i) {
+    i.value()->setVisible(false); 
```

Comment:
I think this is fine, although it looks brute force. If you set visible to false when it's already false then nothing happens in Qt, so you're not adding extra computation here. I think if you added cache variables it would be actually less clean.